scripts/4.Covariates.py

```diff
@@ -67,156 +59,98 @@
 pd.get_dummies(samples_df.dead).head(10)
 
 
-# Note that null values are encoded as (0,0) pairs... I was wondering if we were okay with this.
+# Note that null values are encoded as (0,0) pairs.
 # 
 # Let's rename the values in each of these so that they more accurately reflect the underlying variable.
 
 # In[7]:
 
 mortality_map = pd.Series(['alive', 'dead'], index=[0.0, 1.0])
-mortality = samples_df.dead.map(mortality_map)
-mortality.head(10)
+samples_df.dead = samples_df.dead.map(mortality_map)
+samples_df.dead.head(10)
 
 
 # In[8]:
 
 recurred_map = pd.Series(['has_not_recurred', 'has_recurred'], index=[0.0, 1.0])
-recurred = samples_df.recurred.map(recurred_map)
-recurred.head(10)
+samples_df.recurred = samples_df.recurred.map(recurred_map)
+samples_df.recurred.head(10)
 
 
 # Now when we concatenate, the columns are appropriately named, and the values are encoded as numeric data that can be input to the types of classifiers that we have been using.
 
 # In[9]:
 
-pd.concat([pd.get_dummies(mortality), pd.get_dummies(recurred)], axis=1).head(10)
+pd.get_dummies(samples_df[['dead', 'recurred']]).head(10)
 
 
-# Let's process the entire thing. We'll convert the categorical variables to dummies, and then concatenate the numeric variables (age of diagnosis, days survived, and days recurrence free).
+# Another useful covariate will be the logarithm plus one function of the number mutations that was calculated in the aforementioned notebook.
 
 # In[10]:
 
-columns_to_process = [samples_df.disease,
-                      samples_df.organ_of_origin,
-                      samples_df.gender,
-                      mortality,
-                      recurred]
-columns_as_dummies = [pd.get_dummies(column) for column in columns_to_process]
+path = os.path.join('download', 'mutation-matrix.tsv.bz2')
+mutation_df = pd.read_table(path, index_col=0)
+
+
+# In[11]:
+
+mutation_df.head(2)
+
+
+# In[12]:
+
+number_of_mutations = mutation_df.sum(axis='columns')
+number_of_mutations.head(2)
+
+
+# Let's process the entire thing. We'll convert the categorical variables to dummies, and then concatenate the numeric variables (age of diagnosis, days survived, and days recurrence free).
+
+# In[13]:
+
 numeric_columns = ['age_diagnosed', 'days_survived', 'days_recurrence_free']
-covariates_df = pd.concat(columns_as_dummies+[samples_df[numeric_columns]], axis=1)
+mutation_column = np.log1p(number_of_mutations)
+covariates_df = pd.concat(
+    [samples_df[numeric_columns], pd.get_dummies(samples_df[categorical_variables])],
```

Comment:
Sorry, this is indeed correct! It is not at all clear from the pandas documentation...